### PR TITLE
Allow subdomains for domain_name

### PR DESCRIPTION
**Pull request by thiagofigueiro**
Implements #505 

Please let me know if you'd like tests for this specific functionality.  I had a look at the existing tests and it doesn't look like it's common practice in this project to test parameters.

### CODE REVIEW

File: faker/providers/internet/__init__.py [hunk 1]
```diff
@@ -97,8 +97,20 @@ def user_name(self):
...
+        """
+        Produce an Internet domain name with the specified number of
+        subdomain levels.
+
+        >>> domain_name()
+        nichols-phillips.com
+        >>> domain_name(2)
+        williamson-hopkins.jackson.com
+        """
+        if 1 >= levels:
```
Comment (fcurella) [inline faker/providers/internet/__init__.py]: just a stylistic note: could change this to be `if levels == 1:`

**Comment by fcurella** [comment]
@thiagofigueiro more tests are always good. Feel free to add some :)

File: faker/providers/internet/__init__.py (hunk 1 above)
Comment (thiagofigueiro) [inline faker/providers/internet/__init__.py]: I changed and added a check to avoid an infinite recursion.

**Comment by thiagofigueiro** [comment]
Sorry, @fcurella - do you mind reviewing again?  I added a couple of tests.

**Comment by fcurella** [comment]
looks good! Thank you! ✨

**Pull request merged**